tagcompare/capture.py

```diff
@@ -14,11 +17,164 @@
+            webdriver.screenshot_element(
+                self._driver, tag_element, output_path)
+        except selenium.common.exceptions.WebDriverException:
+            LOGGER.exception("Exception while capturing tags!")
+            return False
+        return errors
+
+    def _capture_tag(self, pathbuilder, tags_per_campaign,
+                     capture_existing=False):
+        """
+        Captures a tag
+        :param pathbuilder:
+        :param tags_per_campaign:
+        :param tagsize:
+        :param capabilities:
+        :param capture_existing:
+        :return: list of browser errors during capture.
+            False on error, None on skip
+        """
+        # Check if we already have the files from default path
+        default_pb = pathbuilder.clone(build=output.DEFAULT_BUILD_NAME)
+        if default_pb.pathexists() and not capture_existing:
+            LOGGER.debug("Skipping existing captures %s", default_pb.path)
+            return None
+
+        tag_html = tags_per_campaign[pathbuilder.tagsize][pathbuilder.tagtype]
+        pathbuilder.create()
+        output_path = pathbuilder.tagimage
+        errors = self.capture_tag(tag_html=tag_html, output_path=output_path,
+                                  tagtype=pathbuilder.tagtype)
+        self.__write_html(
+            tag_html=tag_html, output_path=pathbuilder.taghtml)
+        return errors
+
+    def capture_tags(self, tags, pathbuilder,
+                     tagsizes=settings.DEFAULT.tagsizes,
+                     tagtypes=settings.DEFAULT.tagtypes,
+                     capture_existing=False):
+        num_existing_skipped = 0
+        num_captured = 0
+        browser_errors = []
+
+        try:
+            for cid in tags:
+                pathbuilder.cid = cid
+                tags_per_campaign = tags[cid]
+                # LOGGER.debug("tags_per_campaign: %s", str(tags_per_campaign))
+                # TODO: Refactor better with _capture_tag
+                # It's weird that we pass in a pathbuilder object and do two
+                # nested loops here
+                for tagsize in tagsizes:
+                    if tagsize not in tags_per_campaign:
+                        LOGGER.warn("No tagsize '%s' found for campaign: %s. Skipping",
+                                    tagsize, cid)
+                        continue
+                    pathbuilder.tagsize = tagsize
+                    for tagtype in tagtypes:
+                        pathbuilder.tagtype = tagtype
+                        r = self._capture_tag(pathbuilder=pathbuilder,
+                                              tags_per_campaign=tags_per_campaign,
+                                              capture_existing=capture_existing)
+                        if r is None:
+                            num_existing_skipped += 1
+                        elif r is False:
+                            continue
```

Comment:
Future readers that guess `False` is an error will also wonder why you are doing nothing about it. Perhaps you do just want to log it, but if you do that here, instead of in `_capture_tag`, then the `log(...); continue` says very clearly "I intend to log this error and then continue". The intent is much less clear, split up as it is now.